src/ReactiveDomain.Testing/Messaging/ConcurrentMessageQueue.cs

```diff
@@ -57,7 +58,10 @@ public ConcurrentMessageQueue(string name, IEnumerable<T> items)
         public void AssertEmpty()
         {
             if (!IsEmpty)
-                throw new Exception($" {_name} Queue not Empty.");
+            {
+                var msg = this.ElementAt(0);
+                throw new Exception($" {_name} Queue not Empty. Instead {msg.GetType()} is next");
```

Comment:
This gives us an opportunity to clean up an old error and fix a new one:
```suggestion
                throw new Exception($"{_name} Queue is not Empty. Instead {msg.GetType()} is next.");
```